### PR TITLE
Semantically-Aligned Multimodal Utterance-level (SAMU) pre-training

### PR DESCRIPTION
**Pull request by mhn226**
## What does this PR do?
This PR makes possible a fine-tuning process (similar to this [paper](https://arxiv.org/abs/2205.08180)) of pre-trained speech models such as wav2vec2.0. The main idea is to stack a pooling layer to wav2vec model in order to transform it's frame-level embeddings to a single sentence-level embedding of each utterance. This sentence-level embedding is then projected onto the same space as the text sentence-level embedding of the FROZEN pre-trained [LaBSE](https://huggingface.co/setu4993/LaBSE) model. The objective of the fine-tuning process is to maximize the similarity between speech and text sentence-level embedding of a same ASR data example (can as well be AST data example too).

For this purpose, the PR implements:

- LaBSE HF model, which inherits the new HF interface #2116.
- Attention Pooling similar to this [paper](https://arxiv.org/abs/2008.01077)
- A recipe for fine-tuning wav2vec2 model in the mentioned supervised manner on IWSLT Tamasheq-French low-resource data.
- A recipe for using the fine-tuned wav2vec model for the AST task on this same dataset. Here I try to couple the fine-tuned wav2vec model with mBART or NLLB (#2214 ). Results shall show we can gain some BLEU scores by doing that.

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by mravanelli** [comment]
Thank you @mhn226 for this PR. I reviewed, did some fixes, and finally merged!

**Comment by mhn226** [comment]
> Thank you @mhn226 for this PR. I reviewed, did some fixes, and finally merged!

Thank you so much! I'm glad that this is merged!
